_includes/files.html

```diff
@@ -68,9 +82,17 @@
     
     {% else %}
     
+      {% comment %}
+      <!-- iterate through found_file_array and check if the same file_path
+        already exists.  If the same file_path already exists, do nothing.
+        Otherwise, add the new file_path to the array so that the duplicate
+        file title error check will trigger below.  This is only needed as part
+        of a workaround to ensure the same file is counted twice for duplicate
+        title error trapping. -->
+      {% endcomment %}
       {% for found_file in found_file_array %}
         {% if found_file != file_path %}
-          {% assign found_file_array =  found_file_array | push: file_path %}
+          {% assign found_file_array = found_file_array | push: file_path %}
         {% endif %}
       {% endfor %}
```

Comment:
I'm having trouble understanding this logic.

If `found_file_array` is `['A', 'B', 'C']` and `file_path` is `'C'`, won't it do an iteration for `'A'` and push `'C'`, so `found_file_array` is `['A', 'B', 'C', 'C']`? And then do the same thing when it iterates for `'B'`?

I'd rather not modify an array while we're iterating it because it's hard to reason about what will happen. Unless the developer is very familiar with the implementation details of liquid loop iteration, it's unclear if the loop iterates over elements added to the array *during* the loop.

Reply:
Very valid points here...  Once we had an array with a length > 1, it really doesn't matter because it will throw the error as expected and needed.  However, I can see how modifying an array while iterating over it is not a good practice at all.

So, I've re-worked this logic so that it splits a string into the array and just sets a variable if it is NOT found.  Then appends that "not found" value to the string after the iteration has been complete.

So, I think this is better?  Its definitely more wordy/lines, but should be safer...  Since we are comparing file paths, it would be awesome to just see if the string "contains" the active file, but I really just worry that "contains" wouldn't be safe in ALL situations.  So, I think iterating over an array is the most comprehensive solution.

Fixed in 5a3cf23